larevt/CalibrationDBI/Providers/DBFolder.cxx

```diff
@@ -229,81 +255,639 @@ namespace lariov {
+		std::cout << "First column has wrong type null." << std::endl;
+		throw cet::exception("DBFolder") << "First column has wrong type null.";
+	      }
+	    }
+	    else {
+	      std::cout << "Unrecognized sqlite data type" << std::endl;
+	      throw cet::exception("DBFolder") << "Unrecognized sqlite data type.";
+	    }
+	    firstcol = false;
+	  }
+	}
+      }
+      else if(rc != SQLITE_DONE) {
+	std::cout << "sqlite3_step returned error result = " << rc << std::endl;
+	throw cet::exception("DBFolder") << "sqlite3_step error.";
+      }
+    }
+    if(nrows != data.nrows()) {
+      std::cout << "Wrong number of data rows " << nrows << std::endl;
+      throw cet::exception("DBFolder") << "Wrong number of data rows " << nrows;
+    }
+    if(values.size() != data.nrows() * data.ncols()) {
+      std::cout << "Wrong number of values " << values.size() << std::endl;
+      throw cet::exception("DBFolder") << "Wrong number of values " << values.size();
+    }
+
+    // Delete statement.
+
+    sqlite3_finalize(stmt);
+
+    // Close database.
+
+    sqlite3_close(db);
+
+    // Done.
+
+    return;
+  }
+
+  // Dump dataset by rows.
+
+  void DBFolder::DumpDataset(const DBDataset& data) const
+  {
+    size_t nrows = data.nrows();
+    size_t ncols = data.ncols();
+    std::cout << "Dataset contains " << nrows << " rows and " << ncols << " columns." << std::endl;
+
+    // Begin time.
+
+    std::cout << "IOV start time = " << data.beginTime().DBStamp() << std::endl;
+
+    // End time.
+
+    std::cout << "IOV end time = " << data.endTime().DBStamp() << std::endl;
+
+    // Columnn names.
+
+    const std::vector<std::string>& names = data.colNames();
+    for (size_t c=0; c<ncols; ++c)
+      std::cout << "Column " << c << ", name = " << names[c] << std::endl;
+
+    // Row 3 - column types.
+
+    const std::vector<std::string>& types = data.colTypes();
+    for (size_t c=0; c<ncols; ++c)
+      std::cout << "Column " << c << ", type = " << types[c] << std::endl;
+
+    // Data rows.
+
+    for(size_t row=0; row<nrows; ++row) {
+      std::cout << "\nRow " << row << std::endl;
+      DBDataset::DBRow dbrow = data.getRow(row);
+
+      // Loop over columns.
+
+      for(size_t col=0; col<ncols; ++col) {
+	if(types[col] == "bigint" || types[col] == "integer" || types[col] == "boolean") {
+	  long value = dbrow.getLongData(col);
+	  std::cout << names[col] << " = " << value << std::endl;
+	}
+	else if(types[col] == "real") {
+	  double value = dbrow.getDoubleData(col);
+	  std::cout << names[col] << " = " << value << std::endl;
+	}
+	else if(types[col] == "text" or types[col] == "boolean") {
+	  std::string value = dbrow.getStringData(col);
+	  std::cout << names[col] << " = " << value << std::endl;
+	}
+	else {
+	  std::cout << "Unknown type " << types[col] << std::endl;
+	  throw cet::exception("DBFolder") << "Unknown type.";
+	}
+      }
+    }
+  }
+
+  void DBFolder::CompareDataset(const DBDataset& data1, DBDataset& data2) const
```

Comment:
Suggest returning `bool` so that manual inspection is not necessary--i.e. it would be ideal to type in a test:

```.cpp
assert(dbFolder.CompareDataset(db1, db2));
```
Also, `std::cout` statements should be converted to message-facility ones, and is there a reason the second function argument cannot be marked `const`?